components/series/stackedBar.js

```diff
@@ -3,80 +3,45 @@
 
     fc.series.stackedBar = function() {
 
+        var stackLayout = d3.layout.stack();
+
         var stackedBar = function(selection) {
-            var container;
-
-            // takes an object with values associated with each property, and
-            // converts it into an array of values. Each value has the xValue associated
-            // with it.
-            //
-            // For example, this object:
-            //
-            // obj = { county: 'North Tyneside', labour: 23, conservative: 55 }
-            //
-            // becomes this ...
-            //
-            // [
-            //   { xValue: 'North Tyneside', name: 'labour', previousValue: 0, currentValue: 23},
-            //   { xValue: 'North Tyneside', name: 'conservative', previousValue: 23, currentValue: 78},
-            // ]
-            function objectDatapointToArray(obj) {
-                var values = [];
-                var yTotal = 0;
-                var xVal = obj[stackedBar.xValueKey.value];
-                for (var propertyName in obj) {
-                    if (obj.hasOwnProperty(propertyName) && propertyName !== stackedBar.xValueKey.value) {
-                        var previous = yTotal;
-                        yTotal += Number(obj[propertyName]);
-                        values.push({
-                            'name': propertyName,
-                            'previousValue': previous,
-                            'currentValue': yTotal,
-                            'xValue': xVal
-                        });
-                    }
-                }
-                return values;
-            }
 
             selection.each(function(data) {
 
                 // add a 'root' g element on the first enter selection. This ensures
                 // that it is just added once
-                container = d3.select(this);
+                var container = d3.select(this);
```

Comment:
Yeah - it didn't seem to make sense before the change either. I'll remove it as I'm not sure what it relates to.